tests/unit/test_graph.py

```diff
@@ -109,6 +126,50 @@ def test_transfer(community_with_trustlines):
     assert community.get_account_sum(B).balance == 100
 
 
+def test_triangulation_no_cost(complex_community_with_trustlines_and_fees):
+    complex_community_with_trustlines_and_fees.update_balance(A, B, -10000)  # amount B owes A
+    complex_community_with_trustlines_and_fees.update_balance(A, C, 10000)
+    complex_community_with_trustlines_and_fees.update_balance(B, D, -10000)
+    complex_community_with_trustlines_and_fees.update_balance(C, D, 10000)
+    cost, path = complex_community_with_trustlines_and_fees.find_path_triangulation(
+            A, B, C, 1000)
+    assert path == [A, B, D, C, A]
+    assert cost == 0
+
+
+def test_triangulation_with_cost(complex_community_with_trustlines_and_fees):
+    complex_community_with_trustlines_and_fees.update_balance(A, B, 10000)  # amount B owes A
+    complex_community_with_trustlines_and_fees.update_balance(A, C, -10000)
+    complex_community_with_trustlines_and_fees.update_balance(B, D, 10000)
+    complex_community_with_trustlines_and_fees.update_balance(C, D, -10000)
+    cost, path = complex_community_with_trustlines_and_fees.find_path_triangulation(
+            A, B, C, 1000)
+    assert path == [A, B, D, C, A]
+    assert cost == 44
+
+
+def test_triangulation_negative_value_no_cost(complex_community_with_trustlines_and_fees):
+    complex_community_with_trustlines_and_fees.update_balance(A, B, 10000)  # amount B owes A
+    complex_community_with_trustlines_and_fees.update_balance(A, C, -10000)
+    complex_community_with_trustlines_and_fees.update_balance(B, D, 10000)
+    complex_community_with_trustlines_and_fees.update_balance(C, D, -10000)
+    cost, path = complex_community_with_trustlines_and_fees.find_path_triangulation(
+            A, B, C, -1000)
+    assert path == [A, C, D, B, A]
+    assert cost == 0
+
+
+def test_triangulation_negative_value_with_cost(complex_community_with_trustlines_and_fees):
+    complex_community_with_trustlines_and_fees.update_balance(A, B, -10000)  # amount B owes A
+    complex_community_with_trustlines_and_fees.update_balance(A, C, 10000)
+    complex_community_with_trustlines_and_fees.update_balance(B, D, -10000)
+    complex_community_with_trustlines_and_fees.update_balance(C, D, 10000)
+    cost, path = complex_community_with_trustlines_and_fees.find_path_triangulation(
+            A, B, C, -1000)
+    assert path == [A, C, D, B, A]
+    assert cost == 44
+
```

Comment:
Please add tests that test that the paths are actual doing what they should. Call the find_path_triangulation function and than execute a transfer with that path and check if the balance was reduces/increased by that amount. Most important, test if you can reduce a balance from a value x to zero.